core/modules/system/includes/block.page_components.inc

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * PageComponents extends Block
+ *
+ * This class allows us to display the page title, local task tabs, 
+ * local actions links, and messages in a block.
+ */
+class PageComponents extends Block {
+  /**
+   * {@inheritdoc}
+   */
+  function __construct($plugin_name, array $data) {
+    parent::__construct($plugin_name, $data);
+
+    $this->settings += array(
+      'title_tag' => 'h1',
+      'title_class' => 'page-title',
+      'tab_type' => 'both'
+    );
+  }
+
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    $children = $this->getChildren();
+    $title = $children[$this->childDelta]['info'];
+    return strlen($this->settings['title']) ? check_plain($this->settings['title']) : $title;
+  }
+
+  /**
+   * Returns the rendered content of this block.
+   *
+   * @return string
+   */
+  function getContent() {
+    return theme('page_components', array('child_delta' => $this->childDelta, 'settings' => $this->settings));
+  }
+
+  /**
+   *  Builds the block's configuration form.
+   */
+  function form(&$form, &$form_state) {
+    parent::form($form, $form_state);
+    if ($this->childDelta == 'title_combo' || $this->childDelta == 'title') {
+      $form['title_wrapper'] = array(
+        '#type' => 'fieldset',
+        '#title' => t('Page title'),
+      );
+      $form['title_wrapper']['title_tag'] = array(
+        '#title' => t('Title tag'),
+        '#type' => 'select',
+        '#options' => array(
+          'none' => t('- No tag -'),
+          'h1' => t('h1'),
+          'h2' => t('h2'),
+          'h3' => t('h3'),
+          'h4' => t('h4'),
+          'h5' => t('h5'),
+          'h6' => t('h6'),
+          'div' => t('div'),
+        ),
+        '#default_value' => $this->settings['title_tag'],
+      );
+      $form['title_wrapper']['title_class'] = array(
+        '#title' => t('CSS class to use'),
+        '#type' => 'textfield',
+        '#default_value' => $this->settings['title_class'],
+      );
+    }
+
+    if ($this->childDelta == 'title_combo' || $this->childDelta == 'tabs') {
+      $form['tabs'] = array(
+        '#type' => 'fieldset',
+        '#title' => t('Local task tabs'),
+      );
+      $form['tabs']['tab_type'] = array(
+        '#title' => t('Tabs type'),
+        '#type' => 'select',
+        '#options' => array(
+          'both' => t('Primary and secondary'),
+          'primary' => t('Primary'),
+          'secondary' => t('Secondary'),
```

Comment:
Curious, what about adding an option for `none` here? I know disabling/removing the tabs from the front-end is a common request. This would allow you to place the block for combo or tabs, and then just not display them at all.